### PR TITLE
DE4762

### PR DESCRIPTION
**Pull request by silbermm**
Add eventId to the query so that we can be more specific about the which invoice to grab

### CODE REVIEW

File: Gateway/MinistryPlatform.Translation/Repositories/InvoiceRepository.cs [hunk 1]
```diff
@@ -92,18 +92,20 @@ public bool CreateInvoiceAndDetail(int productId, int? productOptionPriceId, int
             return _ministryPlatformRest.UsingAuthenticationToken(apiToken).Post(new List<MpNestedInvoiceDetail>(new List<MpNestedInvoiceDetail> { invoice })) == 200;
         }
 
-        public Result<MpInvoiceDetail> GetInvoiceDetailsForProductAndCamper(int productId, int camperId)
+        public Result<MpInvoiceDetail> GetInvoiceDetailsForProductAndCamper(int productId, int camperId, int eventId)
         {
-            var apiToken = _apiUserRepository.GetToken();
-            var invoiceDetails = _ministryPlatformRest.UsingAuthenticationToken(apiToken).Search<MpInvoiceDetail>($"Recipient_Contact_ID_Table.[Contact_ID]={camperId} AND Product_ID_Table.[Product_ID]={productId} AND Invoice_Status_ID!={_invoiceCancelled}", "Invoice_ID_Table.[Invoice_ID]");
-            if (invoiceDetails.Any())
+            var apiToken = _apiUserRepository.GetDefaultApiUserToken();
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Translation/Repositories/InvoiceRepository.cs]: Updating deprecated code. 👍

File: Gateway/MinistryPlatform.Translation/Repositories/InvoiceRepository.cs [hunk 2]
```diff
@@ -92,18 +92,20 @@ public bool CreateInvoiceAndDetail(int productId, int? productOptionPriceId, int
...
+            var invoiceDetails = _ministryPlatformRest.UsingAuthenticationToken(apiToken).Search<MpInvoiceDetail>($"Recipient_Contact_ID_Table.[Contact_ID]={camperId} AND Product_ID_Table.[Product_ID]={productId} AND Event_Participant_ID_Table_Event_ID_Table.[Event_ID]={eventId} AND Invoice_Status_ID!={_invoiceCancelled}", "Invoice_ID_Table.[Invoice_ID]");            
+            if (!invoiceDetails.Any())
             {
-                if (invoiceDetails.First() != null)
-                {
-                    return new Result<MpInvoiceDetail>(true, invoiceDetails.First());
-                }
+              return new Result<MpInvoiceDetail>(false, $"No invoice details for camper: {camperId}, product: {productId} and event: {eventId}");
```
Comment (andycanterbury) [inline Gateway/MinistryPlatform.Translation/Repositories/InvoiceRepository.cs]: I like these error messages, I hope they get exposed on the front end?

File: Gateway/crds-angular/Services/CampService.cs [hunk 3]
```diff
@@ -110,17 +110,21 @@ public ProductDTO GetCampProductDetails(int eventId, int camperContactId, string
         {
             var formId = _configurationWrapper.GetConfigIntValue("SummerCampFormID");
             var formFieldId = _configurationWrapper.GetConfigIntValue("SummerCampForm.FinancialAssistance");
-            var me = _contactRepository.GetMyProfile(token);
```
Comment (andycanterbury) [inline Gateway/crds-angular/Services/CampService.cs]: This wasn't used anywhere? Should save some time not to do it, if it wasn't even used! 👍

Comment (silbermm) [reply Gateway/crds-angular/Services/CampService.cs]: yeah, I was surprised to find it wasn't used, but I verified it wasn't and hopefully we can shave off a few milliseconds

**Pull request merged**